### ISSUE TITLE
Random "Error from server (Forbidden): unknown" after 2.6.3 Upgrade

### ISSUE THREAD

**Issue by andrew-landsverk-win** [rancher/rancher#36875]
**Rancher Server Setup**
- Rancher version:  2.6.3
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1 1.20.6
- Proxy/Cert Details: 

**Information about the Cluster**
- Kubernetes version: 1.21.10, 1.20.13
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Imported MicroK8s

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Cluster Owner

   
**Describe the bug**
After upgrading from Rancher 2.6.2 to 2.6.3 started getting "random" errors when using any `kubectl` commands. From the client side, the error is: `Error from server (Forbidden): unknown`. I then went to the Rancher server pods and found the following (I'm not sure if they are related, but they all occur at the same time as the Forbidden error:

```
2022/03/14 13:58:07 [INFO] Creating token for user u-rvgw2
2022/03/14 13:59:15 [ERROR] Error refreshing token principals, skipping: invalid server config. only exactly 1 server is currently supported
2022/03/14 13:59:52 [ERROR] Unknown error: Operation cannot be fulfilled on preferences.management.cattle.io "ns-by-cluster": the object has been modified; please apply your changes to the latest version and try again
2022/03/14 14:00:05 [ERROR] Error during subscribe websocket: close sent
2022/03/14 14:00:05 [ERROR] Unknown error: Operation cannot be fulfilled on preferences.management.cattle.io "cluster": the object has been modified; please apply your changes to the latest version and try again
```

I am tagging @cmurphy and rancher/rancher#36096 just for some extra visibility. As always, please let me know if there's anything I can do to assist or any additional information that I may provide. Thank you!

**To Reproduce**
I'm not 100% certain of the steps, but this `kubectl` client-side error only started showing up after the Rancher upgrade from 2.6.2 to 2.6.3.

**Result**

**Expected Result**

**Screenshots**

**Additional context**

**Comment by cmurphy** [comment]
@andrew-landsverk-win can you tell me more about your auth provider configuration? Are you using OpenLDAP or Active Directory? Have you ever switched auth providers? Was it configured through the UI, through terraform, or some other way? The log message `invalid server config. only exactly 1 server is currently supported` makes it look like something messed up the LDAP config and there are duplicate servers.

**Comment by andrew-landsverk-win** [comment]
That’s a great question :) Initially we were using AD. Later we switched to AzureAD. It’s only every been configured through the UI. The users where we get the intermittent errors are local users however.

**Comment by andrew-landsverk-win** [comment]
Not a problem at all:

Here is the result from the kubectl command:

`
activedirectory
adfs
azuread
freeipa
github
googleoauth
keycloak
keycloakoidc
local
oidc
okta
openldap
ping
shibboleth
`

We are not using Active Directory Domain Services, it's been a long time, but I believe i used the first doc you linked.

Here is the full output from the second command:
`
activedirectory
adfs
azuread true
freeipa
github
googleoauth
keycloak
keycloakoidc
local   true
oidc
okta
openldap
ping
shibboleth
`

I expected to see more output based on your question though, do I need to run a different command?

Thanks!

**Comment by MbolotSuse** [comment]
@andrew-landsverk-win 

Thank you for running those commands. The output between the two commands was consistent - (the authconfig is off for activedirectory and doesn't have the url for the server that you used to use). I was trying to see if there was anything left over from a previous configuration that could be causing issues. Which there wasn't in the auth configs. But this did lead me to another potential cause of your issue. I need to do some extra testing, but that output helped a ton - thanks again.

**Comment by MbolotSuse** [comment]
@andrew-landsverk-win I've been trying to track down this issue without success (unfortunately). I've got a couple of other questions which might help out a bit if you don't mind:

1. How often do you experience these issues? Is there any common characteristics to moments when you see the problems (same time of day, high amounts of user activity, etc.)? Does this happen with any/every user, or just with a select few?

2. If possible, could you try to capture the output of one of the failing kubectl commands with verbosity set to a higher level (`kubectl get pods -v 6` for example)? Level 6 should give me more info (though you can go higher if you want). I would also recommend carefully reviewing the output before posting here (this could include sensitive information). Unfortunately that `unknown` message doesn't give a lot of detail - more info could help me track this down.

3. Do you always see the `[ERROR] Error refreshing token principals, skipping: invalid server config. only exactly 1 server is currently supported` error messages in the logs when this occurs? Or does this only occur sometimes?

4. Do you see any other log messages at the time that this occurs other than what was originally posted in the issue?

Thank you for your help with tracking this down.

**Comment by andrew-landsverk-win** [comment]
@MbolotSuse thanks for the response.

1. I had an ansible job that ran kubectl to make sure certs were updated from a file every 4 hours (ran with our other automations) and it would consistently fail most of the time. I turned off this feature because of the constant failures. I would also see it more rarely than that when running helm commands. I haven't seen a failure with helm for a while now, and I can't seem to reproduce it at this time by running kubectl commands over and over.

Next time it happens, I'll try to capture more information. I may also re-enable the ansible job to see if we can't catch it in the act next week.

Thanks again!

**Comment by wirwolf** [comment]
Also reproduced on 2.6.4 when ui request monitoring info
cluster agent logs
```
time="2022-04-27T11:52:34Z" level=error msg="Error during subscribe websocket: close sent"
time="2022-04-27T11:54:47Z" level=error msg="Error during subscribe websocket: close sent"
```
rancher instance
```
2022/04/27 11:47:10 [ERROR] Error during subscribe write tcp 10.233.64.140:80->10.233.64.111:60188: write: broken pipe
2022/04/27 11:47:19 [ERROR] Error during subscribe write tcp 10.233.64.140:80->10.233.64.111:60184: write: broken pipe
```
Link
https://rancher.********.org/k8s/clusters/c-m-ncsd5qwm/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/api/dashboards/uid/rancher-pod-1

**Comment by wirwolf** [comment]
PS. Ofen random query in UI loading 60 seconds.
![image](https://user-images.githubusercontent.com/8457572/166644496-c127f003-d4e1-4615-82bb-c6d62fe82a07.png)

**Comment by wirwolf** [comment]
Please fix this issue in v2.6.5. I can not use the last stable version(v2.6.4) because page loading is a very long time! And I have many complaints from my developers about unuseable UI

**Comment by samjustus** [comment]
SURE-4778
